internal/testutils/metric_pipeline_builder.go

```diff
@@ -186,18 +185,29 @@ func (b *MetricPipelineBuilder) Build() telemetryv1alpha1.MetricPipeline {
 						Basic: &telemetryv1alpha1.BasicAuthOptions{
 							User: telemetryv1alpha1.ValueType{
 								Value: b.basicAuthUser,
+								ValueFrom: &telemetryv1alpha1.ValueFromSource{
```

Comment:
Does this mean we did not support basic auth with secret references before?

Reply:
In the testutils, yes